datasets/trecqa.py

```diff
@@ -0,0 +1,76 @@
+import os
+
+import torch
+from torchtext.data.dataset import Dataset
+from torchtext.data.example import Example
+from torchtext.data.field import Field
+from torchtext.data.iterator import BucketIterator
+from torchtext.vocab import Vectors
+
+from datasets.idf_utils import get_pairwise_word_to_doc_freq, get_pairwise_overlap_features
+
+
+class TRECQA(Dataset):
+    NAME = 'trecqa'
+    NUM_CLASSES = 2
+    ID_FIELD = Field(sequential=False, tensor_type=torch.FloatTensor, use_vocab=False, batch_first=True)
+    TEXT_FIELD = Field(batch_first=True, tokenize=lambda x: x)  # tokenizer is identity since we already tokenized it to compute external features
+    EXT_FEATS_FIELD = Field(tensor_type=torch.FloatTensor, use_vocab=False, batch_first=True, tokenize=lambda x: x)
+    LABEL_FIELD = Field(sequential=False, use_vocab=False, batch_first=True)
+
+    @staticmethod
+    def sort_key(ex):
+        return len(ex.a)
+
+    def __init__(self, path):
+        """
+        Create a TRECQA dataset instance
+        """
+        fields = [('id', self.ID_FIELD), ('a', self.TEXT_FIELD), ('b', self.TEXT_FIELD), ('ext_feats', self.EXT_FEATS_FIELD), ('label', self.LABEL_FIELD)]
```

Comment:
although the files are called `a.toks` and `b.toks`, I think we should called these fields as questions and answers

Reply:
They are not necessarily question and answer. MP-CNN can be used for semantic text similarity between two sentences in general. I guess I can call them `sentence_1` and `sentence_2` to be more clearer than just `a` and `b`.